Ray.jl/src/function_manager.jl

```diff
@@ -22,25 +22,38 @@
 # function manager holds:
 # local cache of functions (keyed by function id/hash from descriptor)
 # gcs client
-# maybe job id?
+# ~~maybe job id?~~ this is managed by the core worker process
 
-using ray_core_worker_julia_jll: JuliaGcsClient, Exists, Put, Get, JuliaFunctionDescriptor, function_descriptor
+using ray_core_worker_julia_jll: JuliaGcsClient, Exists, Put, Get,
+                                 JuliaFunctionDescriptor, function_descriptor
 
-# XXX: what's the actual namespace to use?  probably set per-job but I dunno.
-const FUNCTION_MANAGER_NAMESPACE = "JuliaFunctions"
+# python uses "fun" for the namespace: https://github.com/beacon-biosignals/ray/blob/dfk%2Fusing-Ray/python/ray/_private/ray_constants.py#L380
+# so "jlfun" seems reasonable
+const FUNCTION_MANAGER_NAMESPACE = "jlfun"
 
 Base.@kwdef struct FunctionManager
     gcs_client::JuliaGcsClient
     functions::Dict{String,Any}
 end
 
-# XXX: we should probably be packaging task-related metadata like the job_id in
-# a task/remotefunction-like struct instead of passing it around like this.
-function_key(fd::JuliaFunctionDescriptor, job_id) = string("RemoteFunction:", job_id, ":", fd.function_hash)
+const FUNCTION_MANAGER = Ref{FunctionManager}()
```

Comment:
this is only accessible to coreworkers that share memory right?

Reply:
yeah, so running under the same process.  may need to protect it with a lock or something (or build the lock into the function manager itself) in case of any async/multithreading shenanigans but I think this is fine for now